libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccount.java

```diff
@@ -363,6 +373,37 @@ public String getThumbnailUrl() {
         return additionalOauthValues;
     }
 
+    /**
+     * Fetches this user's profile photo from the cache.
+     *
+     * @return User's profile photo.
+     */
+    public Bitmap getProfilePhoto() {
+        final File file = getProfilePhotoFile();
+        final BitmapFactory.Options bitmapOptions = new BitmapFactory.Options();
+        bitmapOptions.inPreferredConfig = Bitmap.Config.ARGB_8888;
+        final Bitmap bitmap = BitmapFactory.decodeFile(file.getAbsolutePath(), bitmapOptions);
+        return bitmap;
+    }
+
+    /**
+     * Fetches this user's profile photo from the server and stores it in the cache.
+     */
+    public void downloadProfilePhoto() {
+        final File file = getProfilePhotoFile();
+        final Uri srcUri = Uri.parse(photoUrl);
+        final Uri destUri = Uri.fromFile(file);
+        final DownloadManager.Request downloadReq = new DownloadManager.Request(srcUri);
```

Comment:
`DownloadManager` is a great way to download files, especially media. It abstracts away the need to create threads, priorities, etc.